src/java.base/share/classes/java/util/Iterator.java

```diff
@@ -83,6 +84,7 @@ public interface Iterator<E> {
      * @return the next element in the iteration
      * @throws NoSuchElementException if the iteration has no more elements
      */
+    @SideEffectsOnly("this")
```

Comment:
This actually might eliminate the error in the [`SetIterator.java`](https://github.com/typetools/checker-framework/blob/da7819dd8e2125d8c040c66e8f1b6c0d7e6a10f3/checker/tests/nullness/SetIteratorTest.java#L29) test case.

I'm not sure why the following error should be issued:

```java
    if (edges.get(parentNode) != null) {
      for (String childNode : edges.get(parentNode).keySet()) {
        // :: error: (dereference.of.nullable)
        edges.get(parentNode).toString();
        for (String childNodeEdgeX : edges.get(parentNode).get(childNode)) {
          childrenString += " " + childNode + "(" + childNodeEdgeX + ")";
        }
      }
    }
```